rkt/api_service.go

```diff
@@ -773,13 +781,46 @@ func runAPIService(cmd *cobra.Command, args []string) (exit int) {
 
 	v1alpha.RegisterPublicAPIServer(publicServer, v1AlphaAPIServer)
 
-	go publicServer.Serve(tcpl)
+	for _, l := range listeners {
+		defer l.Close()
+		go publicServer.Serve(l)
+	}
 
-	stderr.Printf("API service running on %v...", flagAPIServiceListenAddr)
+	stderr.Printf("API service running")
 
 	<-exitCh
 
 	stderr.Print("API service exiting...")
 
 	return
 }
+
+// Open API sockets based on command line parameters and
+// the magic environment variable from systemd
+//
+// see sd_listen_fds(3)
+func openAPISockets() ([]net.Listener, error) {
+	listeners := make([]net.Listener, 0, 1)
+
+	if flagSocketListen {
+		fds := activation.Files(true) // Get the socket fds from systemd
+		if fds == nil || len(fds) == 0 {
+			return nil, fmt.Errorf("--socket passed, but we're not started by systemd (no LISTEN_FDS environment variable)")
```

Comment:
I'd shorten this to `return nil, fmt.Errorf("no systemd socket activation descriptors available")`